nifi-nar-bundles/nifi-scripting-bundle/nifi-scripting-processors/src/main/java/org/apache/nifi/processors/script/AbstractScriptProcessor.java

```diff
@@ -248,11 +253,13 @@ protected void setupEngine() {
                 Thread.currentThread().setContextClassLoader(scriptEngineModuleClassLoader);
             }
 
-            scriptEngine = createScriptEngine();
+            ScriptEngine scriptEngine = createScriptEngine();
```

Comment:
Obviously you moved the declaration to be locally scoped because you intended to add a for loop here. But as of right now, it only populates the queue with a single engine instance, so despite having multiple threads consume and re-populate it, the queue will only ever contain 1 or 0 engines for use.